src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,35 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.getSessionFactory().openSession();
```

Comment:
The method call `factory.getSessionFactory()` is incorrect because `factory` is already an instance of `SessionFactory`. You should call `factory.openSession()` instead.